kv/kvserver: TestConsistencyQueueRecomputeStats failed

(initial description superseded by later occurrences)

This just flaked for me in #49751, is it a different issue?

https://teamcity.cockroachdb.com/viewLog.html?buildId=1989879&buildTypeId=Cockroach_UnitTests

```
W200605 04:34:05.123668 636770 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.66s)
        consistency_queue_test.go:572: still have a SysCount of 123004
```

Here's the detail of this latest failure:

```
"I200605 04:34:04.885886 606561 server/status/runtime.go:499  [n3] runtime stats: 419 MiB RSS, 743 goroutines, 104 MiB/196 MiB/175 MiB GO alloc/idle/total, 24 MiB/59 MiB CGO alloc/total, 119.6 CGO/sec, 6.5/1.3 %(u/s)time, 0.0 %gc (0x), 426 KiB/426 KiB (r/w)net\n"}
"I200605 04:34:05.042624 606854 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:123 LastUpdateNanos:1591331585043498080 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 Int
entCount:0 SysBytes:0 SysCount:123000}\n"}
"E200605 04:34:05.042719 606854 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded\n"}
"E200605 04:34:05.042776 606854 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation \"consistencyChecker queue process replica 35\" timed out after 1m0s\n"}
"I200605 04:34:05.120577 604147 util/stop/stopper.go:539  quiescing\n"}
"I200605 04:34:05.120644 637160 util/stop/stopper.go:539  quiescing\n"}
"I200605 04:34:05.121102 637158 util/stop/stopper.go:539  quiescing\n"}
"W200605 04:34:05.121402 606907 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:\n"}
"W200605 04:34:05.121466 606981 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:\n"}
"I200605 04:34:05.121732 637159 util/stop/stopper.go:539  quiescing\n"}
"W200605 04:34:05.121753 606902 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:\n"}
"W200605 04:34:05.121821 607318 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:\n"}
"W200605 04:34:05.121926 607319 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:\n"}
"W200605 04:34:05.122367 606900 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:\n"}
"W200605 04:34:05.122527 606321 gossip/gossip.go:1525  [n3] no incoming or outgoing connections\n"}
"W200605 04:34:05.122606 606764 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:\n"}
"W200605 04:34:05.122852 605918 gossip/gossip.go:1525  [n2] no incoming or outgoing connections\n"}
"W200605 04:34:05.122981 606966 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:\n"}
"W200605 04:34:05.123668 636770 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer\n"}
"    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.66s)\n"}
"        consistency_queue_test.go:572: still have a SysCount of 123004\n"}
```

Looking at the code, the failure in the test "still have a SysCount" occurs before the test server is being shut down, i.e. it's a legitimate error.

@tbg how serious is this?

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1992223&tab=buildLog) on [master@77419519d4d339a42afc14b27751958f84853db6](https://github.com/cockroachdb/cockroach/commits/77419519d4d339a42afc14b27751958f84853db6):

```
I200605 21:32:35.071726 600744 server/status/runtime.go:499  [n1] runtime stats: 415 MiB RSS, 743 goroutines, 97 MiB/201 MiB/172 MiB GO alloc/idle/total, 25 MiB/53 MiB CGO alloc/total, 119.1 CGO/sec, 5.3/1.4 %(u/s)time, 0.0 %gc (0x), 1.1 MiB/1.1 MiB (r/w)net
W200605 21:32:35.095380 600734 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200605 21:32:35.240614 601168 server/status/runtime.go:499  [n2] runtime stats: 415 MiB RSS, 743 goroutines, 109 MiB/190 MiB/174 MiB GO alloc/idle/total, 25 MiB/53 MiB CGO alloc/total, 120.3 CGO/sec, 5.7/1.4 %(u/s)time, 0.0 %gc (0x), 1.2 MiB/1.2 MiB (r/w)net
I200605 21:32:35.355262 601522 server/status/runtime.go:499  [n3] runtime stats: 415 MiB RSS, 743 goroutines, 118 MiB/181 MiB/174 MiB GO alloc/idle/total, 25 MiB/53 MiB CGO alloc/total, 119.1 CGO/sec, 6.3/1.3 %(u/s)time, 0.0 %gc (0x), 1.3 MiB/1.3 MiB (r/w)net
I200605 21:32:45.066955 600740 gossip/gossip.go:577  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 435/81 sent/received, bytes 128447B/16209B sent/received)
  2: 127.0.0.1:32887 (1m0s)
  3: 127.0.0.1:34583 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200605 21:32:45.074045 600744 server/status/runtime.go:499  [n1] runtime stats: 416 MiB RSS, 744 goroutines, 73 MiB/220 MiB/174 MiB GO alloc/idle/total, 25 MiB/53 MiB CGO alloc/total, 120.3 CGO/sec, 7.8/0.8 %(u/s)time, 0.2 %gc (1x), 3.5 MiB/3.5 MiB (r/w)net
W200605 21:32:45.093268 600734 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200605 21:32:45.233816 601164 gossip/gossip.go:577  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:35327 (1m0s: infos 41/232 sent/received, bytes 8097B/68850B sent/received)
gossip server (0/3 cur/max conns, infos 41/232 sent/received, bytes 8097B/68850B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200605 21:32:45.238382 601168 server/status/runtime.go:499  [n2] runtime stats: 416 MiB RSS, 744 goroutines, 82 MiB/215 MiB/174 MiB GO alloc/idle/total, 25 MiB/53 MiB CGO alloc/total, 119.1 CGO/sec, 7.9/0.9 %(u/s)time, 0.2 %gc (1x), 3.4 MiB/3.4 MiB (r/w)net
I200605 21:32:45.352524 601518 gossip/gossip.go:577  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:35327 (1m0s: infos 42/221 sent/received, bytes 8374B/64667B sent/received)
gossip server (0/3 cur/max conns, infos 42/221 sent/received, bytes 8374B/64667B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200605 21:32:45.355239 601522 server/status/runtime.go:499  [n3] runtime stats: 416 MiB RSS, 743 goroutines, 89 MiB/209 MiB/174 MiB GO alloc/idle/total, 25 MiB/53 MiB CGO alloc/total, 120.3 CGO/sec, 7.4/1.0 %(u/s)time, 0.2 %gc (1x), 3.3 MiB/3.3 MiB (r/w)net
I200605 21:32:45.480871 601818 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:123 LastUpdateNanos:1591392705497254017 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200605 21:32:45.481020 601818 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200605 21:32:45.481153 601818 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200605 21:32:45.649724 599197 util/stop/stopper.go:539  quiescing
I200605 21:32:45.649789 632118 util/stop/stopper.go:539  quiescing
I200605 21:32:45.649826 632116 util/stop/stopper.go:539  quiescing
I200605 21:32:45.650062 632117 util/stop/stopper.go:539  quiescing
W200605 21:32:45.650531 631845 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200605 21:32:45.650803 602334 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200605 21:32:45.650888 601876 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200605 21:32:45.650890 601854 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200605 21:32:45.650955 601826 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200605 21:32:45.650987 601612 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200605 21:32:45.651038 602333 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200605 21:32:45.651100 601334 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200605 21:32:45.652548 600852 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200605 21:32:45.652674 601944 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200605 21:32:45.652685 601727 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.56s)
        consistency_queue_test.go:572: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1992809&tab=buildLog) on [master@dfb6e1bee4abc2f2ff82d545686908d983f82872](https://github.com/cockroachdb/cockroach/commits/dfb6e1bee4abc2f2ff82d545686908d983f82872):

```
W200606 05:50:01.561035 571310 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200606 05:50:01.646440 571636 server/status/runtime.go:499  [n2] runtime stats: 380 MiB RSS, 745 goroutines, 59 MiB/176 MiB/120 MiB GO alloc/idle/total, 24 MiB/54 MiB CGO alloc/total, 120.4 CGO/sec, 6.1/1.8 %(u/s)time, 0.0 %gc (1x), 18 MiB/18 MiB (r/w)net
I200606 05:50:01.745397 572013 server/status/runtime.go:499  [n3] runtime stats: 380 MiB RSS, 745 goroutines, 66 MiB/170 MiB/120 MiB GO alloc/idle/total, 24 MiB/54 MiB CGO alloc/total, 120.3 CGO/sec, 5.7/1.8 %(u/s)time, 0.0 %gc (0x), 18 MiB/18 MiB (r/w)net
I200606 05:50:11.494174 571172 gossip/gossip.go:577  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 436/78 sent/received, bytes 128354B/14853B sent/received)
  2: 127.0.0.1:37859 (1m0s)
  3: 127.0.0.1:46881 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200606 05:50:11.497025 571176 server/status/runtime.go:499  [n1] runtime stats: 380 MiB RSS, 745 goroutines, 42 MiB/190 MiB/114 MiB GO alloc/idle/total, 24 MiB/54 MiB CGO alloc/total, 119.3 CGO/sec, 7.0/1.8 %(u/s)time, 0.0 %gc (1x), 13 MiB/13 MiB (r/w)net
W200606 05:50:11.560611 571310 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200606 05:50:11.640208 571616 gossip/gossip.go:577  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:45435 (1m0s: infos 39/227 sent/received, bytes 7193B/67139B sent/received)
gossip server (0/3 cur/max conns, infos 39/227 sent/received, bytes 7193B/67139B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200606 05:50:11.646815 571636 server/status/runtime.go:499  [n2] runtime stats: 381 MiB RSS, 746 goroutines, 51 MiB/184 MiB/114 MiB GO alloc/idle/total, 24 MiB/54 MiB CGO alloc/total, 120.5 CGO/sec, 7.6/1.9 %(u/s)time, 0.0 %gc (1x), 13 MiB/13 MiB (r/w)net
I200606 05:50:11.743928 572009 gossip/gossip.go:577  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:45435 (1m0s: infos 41/223 sent/received, bytes 7922B/65777B sent/received)
gossip server (0/3 cur/max conns, infos 41/223 sent/received, bytes 7922B/65777B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200606 05:50:11.745540 572013 server/status/runtime.go:499  [n3] runtime stats: 381 MiB RSS, 745 goroutines, 59 MiB/178 MiB/114 MiB GO alloc/idle/total, 24 MiB/54 MiB CGO alloc/total, 120.5 CGO/sec, 7.6/1.8 %(u/s)time, 0.0 %gc (1x), 13 MiB/13 MiB (r/w)net
I200606 05:50:11.871443 572343 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:123 LastUpdateNanos:1591422551872541240 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200606 05:50:11.871514 572343 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200606 05:50:11.871564 572343 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200606 05:50:12.025573 569816 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:29876 LastUpdateNanos:1591422611780872018 IntentAge:0 GCBytesAge:0 LiveBytes:376835 LiveCount:-2037 KeyBytes:-99144 KeyCount:-2037 ValBytes:475979 ValCount:-2037 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200606 05:50:12.113877 569816 util/stop/stopper.go:539  quiescing
I200606 05:50:12.113923 602685 util/stop/stopper.go:539  quiescing
I200606 05:50:12.113948 602683 util/stop/stopper.go:539  quiescing
I200606 05:50:12.113974 602684 util/stop/stopper.go:539  quiescing
W200606 05:50:12.114275 572403 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200606 05:50:12.114314 572328 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200606 05:50:12.114347 572704 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200606 05:50:12.114412 571729 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200606 05:50:12.114513 572365 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200606 05:50:12.114529 602349 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200606 05:50:12.114556 571445 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200606 05:50:12.114645 572340 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200606 05:50:12.114710 572703 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200606 05:50:12.114730 572310 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200606 05:50:12.114737 572373 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.17s)
        consistency_queue_test.go:572: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1994338&tab=buildLog) on [master@3bf61fdac388e3807edd15feaf44028c0398c2ec](https://github.com/cockroachdb/cockroach/commits/3bf61fdac388e3807edd15feaf44028c0398c2ec):

```
I200607 05:53:36.522269 569754 server/status/runtime.go:499  [n3] runtime stats: 363 MiB RSS, 743 goroutines, 73 MiB/160 MiB/149 MiB GO alloc/idle/total, 25 MiB/58 MiB CGO alloc/total, 120.7 CGO/sec, 6.1/1.6 %(u/s)time, 0.0 %gc (1x), 20 MiB/20 MiB (r/w)net
I200607 05:53:46.350399 568979 gossip/gossip.go:577  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 437/80 sent/received, bytes 128659B/15757B sent/received)
  3: 127.0.0.1:46791 (1m0s)
  2: 127.0.0.1:37595 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200607 05:53:46.354153 568704 server/status/runtime.go:499  [n1] runtime stats: 363 MiB RSS, 745 goroutines, 86 MiB/150 MiB/149 MiB GO alloc/idle/total, 25 MiB/58 MiB CGO alloc/total, 120.4 CGO/sec, 5.6/1.9 %(u/s)time, 0.0 %gc (0x), 15 MiB/15 MiB (r/w)net
W200607 05:53:46.377987 568349 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200607 05:53:46.439904 569393 gossip/gossip.go:577  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:46565 (1m0s: infos 41/231 sent/received, bytes 8097B/68151B sent/received)
gossip server (0/3 cur/max conns, infos 41/231 sent/received, bytes 8097B/68151B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200607 05:53:46.441708 569413 server/status/runtime.go:499  [n2] runtime stats: 363 MiB RSS, 745 goroutines, 94 MiB/144 MiB/149 MiB GO alloc/idle/total, 25 MiB/58 MiB CGO alloc/total, 120.4 CGO/sec, 6.2/2.0 %(u/s)time, 0.0 %gc (0x), 15 MiB/15 MiB (r/w)net
I200607 05:53:46.520406 569750 gossip/gossip.go:577  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:46565 (1m0s: infos 41/224 sent/received, bytes 7922B/66142B sent/received)
gossip server (0/3 cur/max conns, infos 41/224 sent/received, bytes 7922B/66142B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200607 05:53:46.522558 569754 server/status/runtime.go:499  [n3] runtime stats: 363 MiB RSS, 743 goroutines, 51 MiB/178 MiB/147 MiB GO alloc/idle/total, 25 MiB/58 MiB CGO alloc/total, 120.7 CGO/sec, 6.8/1.9 %(u/s)time, 0.0 %gc (1x), 15 MiB/15 MiB (r/w)net
I200607 05:53:46.621613 569871 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:0 LastUpdateNanos:1591509166622069084 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200607 05:53:46.621691 569871 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200607 05:53:46.621749 569871 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200607 05:53:46.857579 567589 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:29876 LastUpdateNanos:1591509226551145699 IntentAge:0 GCBytesAge:0 LiveBytes:376835 LiveCount:-2037 KeyBytes:-99144 KeyCount:-2037 ValBytes:475979 ValCount:-2037 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200607 05:53:46.917315 567589 util/stop/stopper.go:539  quiescing
I200607 05:53:46.917419 600491 util/stop/stopper.go:539  quiescing
I200607 05:53:46.917469 600489 util/stop/stopper.go:539  quiescing
I200607 05:53:46.917494 600490 util/stop/stopper.go:539  quiescing
W200607 05:53:46.917865 600092 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200607 05:53:46.917904 570091 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 05:53:46.917969 569928 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 05:53:46.918031 569531 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200607 05:53:46.918080 569198 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200607 05:53:46.918266 569825 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 05:53:46.918288 570047 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 05:53:46.918379 570553 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: EOF:
W200607 05:53:46.918412 570026 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: EOF:
W200607 05:53:46.918512 569821 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200607 05:53:46.918550 570552 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
I200607 05:53:46.920295 600449 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:46565: context canceled
I200607 05:53:46.920315 600448 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:37595: context canceled
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=false (61.12s)
        consistency_queue_test.go:572: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1995233&tab=buildLog) on [master@600d3175a4c95878799add123f4fd2d3ee3fb79f](https://github.com/cockroachdb/cockroach/commits/600d3175a4c95878799add123f4fd2d3ee3fb79f):

```
W200607 22:56:27.198744 602194 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200607 22:56:27.346219 602669 server/status/runtime.go:499  [n2] runtime stats: 411 MiB RSS, 743 goroutines, 67 MiB/229 MiB/141 MiB GO alloc/idle/total, 21 MiB/57 MiB CGO alloc/total, 120.4 CGO/sec, 7.1/1.5 %(u/s)time, 0.2 %gc (1x), 578 KiB/578 KiB (r/w)net
I200607 22:56:27.599433 603093 server/status/runtime.go:499  [n3] runtime stats: 412 MiB RSS, 743 goroutines, 73 MiB/223 MiB/141 MiB GO alloc/idle/total, 21 MiB/57 MiB CGO alloc/total, 119.6 CGO/sec, 6.5/1.4 %(u/s)time, 0.0 %gc (0x), 650 KiB/650 KiB (r/w)net
I200607 22:56:37.124982 600991 gossip/gossip.go:577  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 437/79 sent/received, bytes 128042B/15305B sent/received)
  2: 127.0.0.1:45049 (1m0s)
  3: 127.0.0.1:37573 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200607 22:56:37.136955 602163 server/status/runtime.go:499  [n1] runtime stats: 412 MiB RSS, 743 goroutines, 86 MiB/213 MiB/141 MiB GO alloc/idle/total, 21 MiB/57 MiB CGO alloc/total, 119.0 CGO/sec, 6.7/1.5 %(u/s)time, 0.0 %gc (0x), 1.8 MiB/1.8 MiB (r/w)net
W200607 22:56:37.210251 602194 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200607 22:56:37.342237 602665 gossip/gossip.go:577  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:35969 (1m0s: infos 41/233 sent/received, bytes 8097B/68665B sent/received)
gossip server (0/3 cur/max conns, infos 41/233 sent/received, bytes 8097B/68665B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200607 22:56:37.348047 602669 server/status/runtime.go:499  [n2] runtime stats: 412 MiB RSS, 744 goroutines, 61 MiB/233 MiB/144 MiB GO alloc/idle/total, 21 MiB/57 MiB CGO alloc/total, 119.7 CGO/sec, 8.7/1.6 %(u/s)time, 0.0 %gc (1x), 2.0 MiB/2.0 MiB (r/w)net
I200607 22:56:37.587676 603089 gossip/gossip.go:577  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:35969 (1m0s: infos 40/220 sent/received, bytes 7470B/64193B sent/received)
gossip server (0/3 cur/max conns, infos 40/220 sent/received, bytes 7470B/64193B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200607 22:56:37.590778 603093 server/status/runtime.go:499  [n3] runtime stats: 412 MiB RSS, 743 goroutines, 72 MiB/223 MiB/144 MiB GO alloc/idle/total, 21 MiB/57 MiB CGO alloc/total, 119.2 CGO/sec, 9.4/1.7 %(u/s)time, 0.0 %gc (1x), 2.1 MiB/2.1 MiB (r/w)net
I200607 22:56:37.717212 603308 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:123 LastUpdateNanos:1591570537720477428 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200607 22:56:37.717328 603308 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200607 22:56:37.717389 603308 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200607 22:56:37.975676 600695 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:29876 LastUpdateNanos:1591570597647039978 IntentAge:0 GCBytesAge:0 LiveBytes:376835 LiveCount:-2037 KeyBytes:-99144 KeyCount:-2037 ValBytes:475979 ValCount:-2037 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200607 22:56:38.110352 600695 util/stop/stopper.go:539  quiescing
I200607 22:56:38.110417 633664 util/stop/stopper.go:539  quiescing
I200607 22:56:38.110577 633662 util/stop/stopper.go:539  quiescing
I200607 22:56:38.110721 633663 util/stop/stopper.go:539  quiescing
W200607 22:56:38.116193 603419 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 22:56:38.116786 603485 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 22:56:38.116955 602890 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200607 22:56:38.116985 603311 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 22:56:38.117066 603934 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 22:56:38.117082 603314 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200607 22:56:38.117154 603933 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200607 22:56:38.117258 602462 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200607 22:56:38.117754 633380 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200607 22:56:38.117780 603508 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200607 22:56:38.117842 603392 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.83s)
        consistency_queue_test.go:572: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1995796&tab=buildLog) on [master@b86ced274f4458192c0e2f33ede84e7456861b01](https://github.com/cockroachdb/cockroach/commits/b86ced274f4458192c0e2f33ede84e7456861b01):

```
W200608 05:48:29.199760 581953 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200608 05:48:29.248550 583268 server/status/runtime.go:499  [n2] runtime stats: 393 MiB RSS, 745 goroutines, 85 MiB/151 MiB/132 MiB GO alloc/idle/total, 23 MiB/62 MiB CGO alloc/total, 119.7 CGO/sec, 6.1/1.7 %(u/s)time, 0.0 %gc (0x), 15 MiB/15 MiB (r/w)net
I200608 05:48:29.331308 583618 server/status/runtime.go:499  [n3] runtime stats: 394 MiB RSS, 745 goroutines, 91 MiB/146 MiB/135 MiB GO alloc/idle/total, 23 MiB/62 MiB CGO alloc/total, 119.7 CGO/sec, 6.1/1.6 %(u/s)time, 0.0 %gc (0x), 15 MiB/15 MiB (r/w)net
I200608 05:48:39.146745 582791 gossip/gossip.go:577  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 432/78 sent/received, bytes 126899B/14869B sent/received)
  2: 127.0.0.1:45803 (1m0s)
  3: 127.0.0.1:40129 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200608 05:48:39.148319 582795 server/status/runtime.go:499  [n1] runtime stats: 394 MiB RSS, 745 goroutines, 55 MiB/180 MiB/129 MiB GO alloc/idle/total, 23 MiB/62 MiB CGO alloc/total, 119.7 CGO/sec, 8.0/1.8 %(u/s)time, 0.0 %gc (1x), 11 MiB/11 MiB (r/w)net
W200608 05:48:39.191205 581953 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200608 05:48:39.247001 583248 gossip/gossip.go:577  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:42437 (1m0s: infos 39/226 sent/received, bytes 7209B/67216B sent/received)
gossip server (0/3 cur/max conns, infos 39/226 sent/received, bytes 7209B/67216B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200608 05:48:39.248876 583268 server/status/runtime.go:499  [n2] runtime stats: 394 MiB RSS, 746 goroutines, 63 MiB/173 MiB/129 MiB GO alloc/idle/total, 23 MiB/62 MiB CGO alloc/total, 119.7 CGO/sec, 8.3/1.9 %(u/s)time, 0.0 %gc (1x), 11 MiB/11 MiB (r/w)net
I200608 05:48:39.329526 583614 gossip/gossip.go:577  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:42437 (1m0s: infos 41/220 sent/received, bytes 7922B/64833B sent/received)
gossip server (0/3 cur/max conns, infos 41/220 sent/received, bytes 7922B/64833B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200608 05:48:39.331277 583618 server/status/runtime.go:499  [n3] runtime stats: 394 MiB RSS, 745 goroutines, 70 MiB/168 MiB/129 MiB GO alloc/idle/total, 23 MiB/62 MiB CGO alloc/total, 120.3 CGO/sec, 8.4/1.9 %(u/s)time, 0.0 %gc (1x), 11 MiB/11 MiB (r/w)net
I200608 05:48:39.466894 583723 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:0 LastUpdateNanos:1591595259468289459 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200608 05:48:39.466977 583723 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200608 05:48:39.467029 583723 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200608 05:48:39.555612 581274 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:29876 LastUpdateNanos:1591595319372617305 IntentAge:0 GCBytesAge:0 LiveBytes:376835 LiveCount:-2037 KeyBytes:-99144 KeyCount:-2037 ValBytes:475979 ValCount:-2037 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200608 05:48:39.679533 581274 util/stop/stopper.go:539  quiescing
I200608 05:48:39.679575 614347 util/stop/stopper.go:539  quiescing
I200608 05:48:39.679655 614345 util/stop/stopper.go:539  quiescing
I200608 05:48:39.679709 614346 util/stop/stopper.go:539  quiescing
W200608 05:48:39.679929 583399 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200608 05:48:39.679960 583914 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: EOF:
W200608 05:48:39.679971 583931 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200608 05:48:39.679996 583719 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200608 05:48:39.680053 583988 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: EOF:
W200608 05:48:39.680060 584470 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200608 05:48:39.680076 583746 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200608 05:48:39.680212 583881 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200608 05:48:39.680255 584471 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200608 05:48:39.680369 613993 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200608 05:48:39.680400 582942 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=false (61.05s)
        consistency_queue_test.go:576: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1997096&tab=buildLog) on [master@769897ab419d6f5ab43b38cdbca1872f91642eb5](https://github.com/cockroachdb/cockroach/commits/769897ab419d6f5ab43b38cdbca1872f91642eb5):

```
  n2 -> n1; n3 -> n1;
I200608 17:58:56.957697 624273 server/status/runtime.go:499  [n2] runtime stats: 423 MiB RSS, 746 goroutines, 80 MiB/221 MiB/144 MiB GO alloc/idle/total, 24 MiB/62 MiB CGO alloc/total, 119.8 CGO/sec, 7.4/1.2 %(u/s)time, 0.3 %gc (1x), 1.2 MiB/1.2 MiB (r/w)net
I200608 17:58:57.061410 624643 gossip/gossip.go:577  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:41951 (1m0s: infos 39/220 sent/received, bytes 7020B/64489B sent/received)
gossip server (0/3 cur/max conns, infos 39/220 sent/received, bytes 7020B/64489B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200608 17:58:57.065387 624647 server/status/runtime.go:499  [n3] runtime stats: 423 MiB RSS, 743 goroutines, 67 MiB/227 MiB/144 MiB GO alloc/idle/total, 24 MiB/62 MiB CGO alloc/total, 120.0 CGO/sec, 8.1/1.2 %(u/s)time, 0.3 %gc (1x), 1.2 MiB/1.2 MiB (r/w)net
I200608 17:58:57.234080 624946 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:123 LastUpdateNanos:1591639077235993187 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200608 17:58:57.234189 624946 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200608 17:58:57.234244 624946 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200608 17:58:57.433949 622375 util/stop/stopper.go:539  quiescing
I200608 17:58:57.434014 655325 util/stop/stopper.go:539  quiescing
I200608 17:58:57.434143 655323 util/stop/stopper.go:539  quiescing
I200608 17:58:57.434254 655324 util/stop/stopper.go:539  quiescing
W200608 17:58:57.442033 624456 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200608 17:58:57.442161 624943 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200608 17:58:57.442329 625008 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200608 17:58:57.446399 624054 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200608 17:58:57.446607 624937 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200608 17:58:57.446840 625524 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200608 17:58:57.446992 654990 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200608 17:58:57.447286 624679 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200608 17:58:57.447338 624968 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200608 17:58:57.447817 625523 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200608 17:58:57.447930 624666 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200608 17:58:57.451708 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200608 17:58:57.451731 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200608 17:58:57.451743 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (split): node unavailable; try another peer
I200608 17:58:57.451754 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200608 17:58:57.451773 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200608 17:58:57.451783 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200608 17:58:57.451794 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200608 17:58:57.451804 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200608 17:58:57.451815 623796 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200608 17:58:57.463590 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200608 17:58:57.463621 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200608 17:58:57.463633 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (split): node unavailable; try another peer
I200608 17:58:57.463646 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200608 17:58:57.463658 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200608 17:58:57.463669 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200608 17:58:57.463681 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200608 17:58:57.463692 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200608 17:58:57.463703 624661 kv/kvserver/queue.go:580  [n3,s3] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200608 17:58:57.463940 655080 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45155: context canceled
I200608 17:58:57.464046 655079 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41951: context canceled
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.41s)
        consistency_queue_test.go:576: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1999198&tab=buildLog) on [master@3c756942000f7c2292ed1c3e68f78aa1d9f33dd2](https://github.com/cockroachdb/cockroach/commits/3c756942000f7c2292ed1c3e68f78aa1d9f33dd2):

```
W200609 05:46:09.443012 582147 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200609 05:46:09.495486 582524 server/status/runtime.go:504  [n2] runtime stats: 391 MiB RSS, 791 goroutines, 70 MiB/160 MiB/162 MiB GO alloc/idle/total, 24 MiB/59 MiB CGO alloc/total, 119.5 CGO/sec, 7.7/2.0 %(u/s)time, 0.0 %gc (1x), 13 MiB/13 MiB (r/w)net
I200609 05:46:09.586914 582907 server/status/runtime.go:504  [n3] runtime stats: 391 MiB RSS, 791 goroutines, 76 MiB/154 MiB/162 MiB GO alloc/idle/total, 24 MiB/59 MiB CGO alloc/total, 120.1 CGO/sec, 7.6/2.1 %(u/s)time, 0.0 %gc (1x), 13 MiB/13 MiB (r/w)net
I200609 05:46:19.387296 581970 gossip/gossip.go:578  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 431/79 sent/received, bytes 126114B/15305B sent/received)
  2: 127.0.0.1:32783 (1m0s)
  3: 127.0.0.1:40159 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200609 05:46:19.389886 582086 server/status/runtime.go:504  [n1] runtime stats: 391 MiB RSS, 791 goroutines, 90 MiB/144 MiB/162 MiB GO alloc/idle/total, 24 MiB/59 MiB CGO alloc/total, 120.7 CGO/sec, 9.3/2.4 %(u/s)time, 0.0 %gc (1x), 232 KiB/232 KiB (r/w)net
W200609 05:46:19.437631 582147 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200609 05:46:19.493955 582520 gossip/gossip.go:578  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:44961 (1m0s: infos 39/225 sent/received, bytes 7193B/66423B sent/received)
gossip server (0/3 cur/max conns, infos 39/225 sent/received, bytes 7193B/66423B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200609 05:46:19.495943 582524 server/status/runtime.go:504  [n2] runtime stats: 392 MiB RSS, 792 goroutines, 98 MiB/138 MiB/162 MiB GO alloc/idle/total, 24 MiB/59 MiB CGO alloc/total, 120.3 CGO/sec, 8.3/2.3 %(u/s)time, 0.0 %gc (0x), 238 KiB/238 KiB (r/w)net
I200609 05:46:19.585133 582903 gossip/gossip.go:578  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:44961 (1m0s: infos 42/220 sent/received, bytes 8374B/64253B sent/received)
gossip server (0/3 cur/max conns, infos 42/220 sent/received, bytes 8374B/64253B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200609 05:46:19.587010 582907 server/status/runtime.go:504  [n3] runtime stats: 392 MiB RSS, 791 goroutines, 105 MiB/132 MiB/162 MiB GO alloc/idle/total, 24 MiB/59 MiB CGO alloc/total, 120.3 CGO/sec, 8.3/2.3 %(u/s)time, 0.0 %gc (0x), 244 KiB/244 KiB (r/w)net
I200609 05:46:19.679910 583102 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:0 LastUpdateNanos:1591681519682348368 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200609 05:46:19.680020 583102 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200609 05:46:19.680068 583102 kv/kvserver/queue.go:1080  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200609 05:46:19.776934 580680 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:29876 LastUpdateNanos:1591681579610270266 IntentAge:0 GCBytesAge:0 LiveBytes:376835 LiveCount:-2037 KeyBytes:-99144 KeyCount:-2037 ValBytes:475979 ValCount:-2037 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200609 05:46:19.976017 580680 util/stop/stopper.go:539  quiescing
I200609 05:46:19.976074 613688 util/stop/stopper.go:539  quiescing
I200609 05:46:19.976098 613686 util/stop/stopper.go:539  quiescing
I200609 05:46:19.976126 613687 util/stop/stopper.go:539  quiescing
W200609 05:46:19.976404 583243 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200609 05:46:19.976448 613299 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200609 05:46:19.976508 583219 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200609 05:46:19.976504 583235 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200609 05:46:19.976553 583270 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200609 05:46:19.976645 583192 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200609 05:46:19.976663 582267 gossip/gossip.go:1528  [n2] no incoming or outgoing connections
W200609 05:46:19.976695 582747 gossip/gossip.go:1528  [n3] no incoming or outgoing connections
W200609 05:46:19.976806 583179 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200609 05:46:19.976849 583704 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200609 05:46:19.976895 583705 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=false (61.20s)
        consistency_queue_test.go:576: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2002171&tab=buildLog) on [master@c5e5f5f88a6b5dfb89585e53a0a39414b225717c](https://github.com/cockroachdb/cockroach/commits/c5e5f5f88a6b5dfb89585e53a0a39414b225717c):

```
W200610 05:59:50.815861 625806 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200610 05:59:50.890259 626234 server/status/runtime.go:504  [n2] runtime stats: 386 MiB RSS, 791 goroutines, 134 MiB/92 MiB/226 MiB GO alloc/idle/total, 24 MiB/58 MiB CGO alloc/total, 120.3 CGO/sec, 5.7/1.2 %(u/s)time, 0.0 %gc (0x), 32 MiB/32 MiB (r/w)net
I200610 05:59:50.982936 626644 server/status/runtime.go:504  [n3] runtime stats: 386 MiB RSS, 791 goroutines, 142 MiB/85 MiB/226 MiB GO alloc/idle/total, 24 MiB/58 MiB CGO alloc/total, 120.3 CGO/sec, 5.7/1.3 %(u/s)time, 0.0 %gc (0x), 32 MiB/32 MiB (r/w)net
I200610 06:00:00.793280 625638 gossip/gossip.go:578  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 426/77 sent/received, bytes 124361B/14403B sent/received)
  2: 127.0.0.1:34443 (1m0s)
  3: 127.0.0.1:38155 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200610 06:00:00.795633 625642 server/status/runtime.go:504  [n1] runtime stats: 386 MiB RSS, 791 goroutines, 156 MiB/73 MiB/226 MiB GO alloc/idle/total, 24 MiB/58 MiB CGO alloc/total, 120.3 CGO/sec, 7.1/1.4 %(u/s)time, 0.0 %gc (0x), 971 KiB/971 KiB (r/w)net
W200610 06:00:00.812605 625806 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200610 06:00:00.888115 626230 gossip/gossip.go:578  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:41325 (1m0s: infos 40/220 sent/received, bytes 7645B/64788B sent/received)
gossip server (0/3 cur/max conns, infos 40/220 sent/received, bytes 7645B/64788B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200610 06:00:00.890631 626234 server/status/runtime.go:504  [n2] runtime stats: 387 MiB RSS, 792 goroutines, 98 MiB/118 MiB/226 MiB GO alloc/idle/total, 24 MiB/58 MiB CGO alloc/total, 120.3 CGO/sec, 9.3/1.5 %(u/s)time, 0.0 %gc (1x), 824 KiB/824 KiB (r/w)net
I200610 06:00:00.981066 626624 gossip/gossip.go:578  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:41325 (1m0s: infos 39/216 sent/received, bytes 7020B/63039B sent/received)
gossip server (0/3 cur/max conns, infos 39/216 sent/received, bytes 7020B/63039B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200610 06:00:00.982814 626644 server/status/runtime.go:504  [n3] runtime stats: 387 MiB RSS, 791 goroutines, 109 MiB/110 MiB/226 MiB GO alloc/idle/total, 24 MiB/58 MiB CGO alloc/total, 120.3 CGO/sec, 9.5/1.4 %(u/s)time, 0.0 %gc (1x), 711 KiB/711 KiB (r/w)net
I200610 06:00:01.024046 626803 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:123 LastUpdateNanos:1591768741054409159 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200610 06:00:01.024164 626803 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200610 06:00:01.024208 626803 kv/kvserver/queue.go:1082  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200610 06:00:01.286208 624358 util/stop/stopper.go:539  quiescing
I200610 06:00:01.286254 657393 util/stop/stopper.go:539  quiescing
I200610 06:00:01.286282 657391 util/stop/stopper.go:539  quiescing
I200610 06:00:01.286327 657392 util/stop/stopper.go:539  quiescing
W200610 06:00:01.286402 656840 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/2/0: node unavailable; try another peer
W200610 06:00:01.286732 626750 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.286759 626891 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.286813 626689 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.286892 626906 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.286973 626439 gossip/gossip.go:1528  [n3] no incoming or outgoing connections
W200610 06:00:01.287337 627238 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.287352 626969 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.287404 626915 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.287406 627239 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200610 06:00:01.287527 626018 gossip/gossip.go:1528  [n2] no incoming or outgoing connections
I200610 06:00:01.287738 657352 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38155: context canceled
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.10s)
        consistency_queue_test.go:580: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestConsistencyQueueRecomputeStats failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2005064&tab=buildLog) on [master@49339fef48f4165a4e7471290bb08d422cf835a9](https://github.com/cockroachdb/cockroach/commits/49339fef48f4165a4e7471290bb08d422cf835a9):

```
W200611 05:51:12.393917 578729 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200611 05:51:12.476078 579192 server/status/runtime.go:504  [n2] runtime stats: 372 MiB RSS, 791 goroutines, 88 MiB/138 MiB/200 MiB GO alloc/idle/total, 25 MiB/66 MiB CGO alloc/total, 119.7 CGO/sec, 6.7/1.2 %(u/s)time, 0.0 %gc (1x), 28 MiB/28 MiB (r/w)net
I200611 05:51:12.550069 579564 server/status/runtime.go:504  [n3] runtime stats: 373 MiB RSS, 791 goroutines, 96 MiB/131 MiB/200 MiB GO alloc/idle/total, 25 MiB/66 MiB CGO alloc/total, 120.3 CGO/sec, 6.7/1.3 %(u/s)time, 0.0 %gc (1x), 28 MiB/28 MiB (r/w)net
I200611 05:51:22.369048 578740 gossip/gossip.go:578  [n1] gossip status (ok, 3 nodes)
gossip client (0/3 cur/max conns)
gossip server (2/3 cur/max conns, infos 442/80 sent/received, bytes 130032B/15757B sent/received)
  2: 127.0.0.1:38289 (1m0s)
  3: 127.0.0.1:43865 (1m0s)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200611 05:51:22.371098 578744 server/status/runtime.go:504  [n1] runtime stats: 373 MiB RSS, 791 goroutines, 109 MiB/121 MiB/200 MiB GO alloc/idle/total, 25 MiB/66 MiB CGO alloc/total, 120.4 CGO/sec, 7.4/1.4 %(u/s)time, 0.0 %gc (1x), 20 MiB/20 MiB (r/w)net
W200611 05:51:22.387735 578729 server/node.go:747  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:34}]}
I200611 05:51:22.473719 579188 gossip/gossip.go:578  [n2] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:45059 (1m0s: infos 40/232 sent/received, bytes 7645B/68760B sent/received)
gossip server (0/3 cur/max conns, infos 40/232 sent/received, bytes 7645B/68760B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200611 05:51:22.476133 579192 server/status/runtime.go:504  [n2] runtime stats: 373 MiB RSS, 792 goroutines, 118 MiB/115 MiB/200 MiB GO alloc/idle/total, 25 MiB/66 MiB CGO alloc/total, 120.4 CGO/sec, 6.1/1.5 %(u/s)time, 0.0 %gc (0x), 20 MiB/20 MiB (r/w)net
I200611 05:51:22.547209 579560 gossip/gossip.go:578  [n3] gossip status (ok, 3 nodes)
gossip client (1/3 cur/max conns)
  1: 127.0.0.1:45059 (1m0s: infos 42/224 sent/received, bytes 8374B/66422B sent/received)
gossip server (0/3 cur/max conns, infos 42/224 sent/received, bytes 8374B/66422B sent/received)
gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I200611 05:51:22.549232 579564 server/status/runtime.go:504  [n3] runtime stats: 374 MiB RSS, 791 goroutines, 125 MiB/109 MiB/200 MiB GO alloc/idle/total, 25 MiB/66 MiB CGO alloc/total, 120.4 CGO/sec, 6.1/1.5 %(u/s)time, 0.0 %gc (0x), 20 MiB/20 MiB (r/w)net
I200611 05:51:22.632807 579770 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] triggering stats recomputation to resolve delta of {ContainsEstimates:123 LastUpdateNanos:1591854622630176111 IntentAge:0 GCBytesAge:0 LiveBytes:0 LiveCount:0 KeyBytes:0 KeyCount:0 ValBytes:0 ValCount:0 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:123000}
E200611 05:51:22.632955 579770 kv/kvserver/consistency_queue.go:147  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] aborted in distSender: context deadline exceeded
E200611 05:51:22.633011 579770 kv/kvserver/queue.go:1082  [n1,consistencyChecker,s1,r35/1:{a-/Table/Syste…}] operation "consistencyChecker queue process replica 35" timed out after 1m0s
I200611 05:51:22.769050 577249 kv/kvserver/replica_consistency.go:246  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:29876 LastUpdateNanos:1591854682583670791 IntentAge:0 GCBytesAge:0 LiveBytes:376835 LiveCount:-2037 KeyBytes:-99144 KeyCount:-2037 ValBytes:475979 ValCount:-2037 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200611 05:51:22.857235 577249 util/stop/stopper.go:539  quiescing
I200611 05:51:22.857286 610300 util/stop/stopper.go:539  quiescing
I200611 05:51:22.857313 610298 util/stop/stopper.go:539  quiescing
I200611 05:51:22.857335 610299 util/stop/stopper.go:539  quiescing
W200611 05:51:22.857700 579789 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200611 05:51:22.857746 579349 gossip/gossip.go:1528  [n3] no incoming or outgoing connections
W200611 05:51:22.857805 610006 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
W200611 05:51:22.857848 579710 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200611 05:51:22.857871 580385 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200611 05:51:22.857976 580384 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200611 05:51:22.858403 579867 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200611 05:51:22.858559 579729 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200611 05:51:22.858790 578954 gossip/gossip.go:1528  [n2] no incoming or outgoing connections
W200611 05:51:22.859554 579707 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200611 05:51:22.859730 579762 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
    --- FAIL: TestConsistencyQueueRecomputeStats/hadEstimates=true (61.12s)
        consistency_queue_test.go:580: still have a SysCount of 123004

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestConsistencyQueueRecomputeStats PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConsistencyQueueRecomputeStats.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/consistency_queue_test.go
+++ b/pkg/kv/kvserver/consistency_queue_test.go
@@ -391,7 +391,6 @@ func TestCheckConsistencyInconsistent(t *testing.T) {
 // The upreplication here is immaterial and serves only to add realism to the test.
 func TestConsistencyQueueRecomputeStats(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("Skip until #48251 is fixed")
 	testutils.RunTrueAndFalse(t, "hadEstimates", testConsistencyQueueRecomputeStatsImpl)
 }
 

--- a/pkg/kv/kvserver/replica_consistency.go
+++ b/pkg/kv/kvserver/replica_consistency.go
@@ -423,16 +423,21 @@ func (r *Replica) getChecksum(ctx context.Context, id uuid.UUID) (ReplicaChecksu
 		r.mu.checksums[id] = c
 	}
 	r.mu.Unlock()
-	// Wait
-	select {
-	case <-r.store.Stopper().ShouldQuiesce():
-		return ReplicaChecksum{},
-			errors.Errorf("store quiescing while waiting for compute checksum (ID = %s)", id)
-	case <-ctx.Done():
-		return ReplicaChecksum{},
-			errors.Wrapf(ctx.Err(), "while waiting for compute checksum (ID = %s)", id)
-	case <-c.notify:
+
+	// Wait for the checksum to compute or at least to start.
+	computed, err := r.checksumInitialWait(ctx, id, c.notify)
+	if err != nil {
+		return ReplicaChecksum{}, err
 	}
+	// If the checksum started, but has not completed commit
+	// to waiting the full deadline.
+	if !computed {
+		_, err = r.checksumWait(ctx, id, c.notify, nil)
+		if err != nil {
+			return ReplicaChecksum{}, err
+		}
+	}
+
 	if log.V(1) {
 		log.Infof(ctx, "waited for compute checksum for %s", timeutil.Since(now))
 	}
@@ -446,6 +451,61 @@ func (r *Replica) getChecksum(ctx context.Context, id uuid.UUID) (ReplicaChecksu
 		return ReplicaChecksum{}, errors.Errorf("no checksum found (ID = %s)", id)
 	}
 	return c, nil
+}
+
+// Waits for the checksum to be available or for the checksum to start computing.
+// If we waited for 10% of the deadline and it has not started, then it's
+// unlikely to start because this replica is most likely being restored from
+// snapshots.
+func (r *Replica) checksumInitialWait(
+	ctx context.Context, id uuid.UUID, notify chan struct{},
+) (bool, error) {
+	d, dOk := ctx.Deadline()
+	// The max wait time should be 5 seconds, so we dont end up waiting for
+	// minutes for a huge range.
+	maxInitialWait := 5 * time.Second
+	var initialWait <-chan time.Time
+	if dOk {
+		duration := time.Duration(timeutil.Until(d).Nanoseconds() / 10)
+		if duration > maxInitialWait {
+			duration = maxInitialWait
+		}
+		initialWait = time.After(duration)
+	} else {
+		initialWait = time.After(maxInitialWait)
+	}
+	return r.checksumWait(ctx, id, notify, initialWait)
+}
+
+// checksumWait waits for the checksum to be available or for the computation
+// to start  within the initialWait time. The bool return flag is used to
+// indicate if a checksum is available (true) or if the initial wait has expired
+// and the caller should wait more, since the checksum computation started.
+func (r *Replica) checksumWait(
+	ctx context.Context, id uuid.UUID, notify chan struct{}, initialWait <-chan time.Time,
+) (bool, error) {
+	// Wait
+	select {
+	case <-r.store.Stopper().ShouldQuiesce():
+		return false,
+			errors.Errorf("store quiescing while waiting for compute checksum (ID = %s)", id)
+	case <-ctx.Done():
+		return false,
+			errors.Wrapf(ctx.Err(), "while waiting for compute checksum (ID = %s)", id)
+	case <-initialWait:
+		{
+			r.mu.Lock()
+			started := r.mu.checksums[id].started
+			r.mu.Unlock()
+			if !started {
+				return false,
+					errors.Errorf("checksum computation did not start in time for (ID = %s)", id)
+			}
+			return false, nil
+		}
+	case <-notify:
+		return true, nil
+	}
 }
 
 // computeChecksumDone adds the computed checksum, sets a deadline for GCing the

--- a/pkg/kv/kvserver/replica_consistency_test.go
+++ b/pkg/kv/kvserver/replica_consistency_test.go
@@ -13,6 +13,7 @@ package kvserver
 import (
 	"context"
 	"testing"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
@@ -55,4 +56,63 @@ func TestReplicaChecksumVersion(t *testing.T) {
 			require.NotNil(t, rc.Checksum)
 		}
 	})
+}
+
+func TestGetChecksumNotSuccessfulExitConditions(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1000*time.Millisecond)
+	defer cancel()
+
+	tc := testContext{}
+	stopper := stop.NewStopper()
+	defer stopper.Stop(ctx)
+	tc.Start(t, stopper)
+
+	id := uuid.FastMakeV4()
+	notify := make(chan struct{})
+	close(notify)
+
+	// Simple condition, the checksum is notified, but not computed.
+	tc.repl.mu.Lock()
+	tc.repl.mu.checksums[id] = ReplicaChecksum{notify: notify}
+	tc.repl.mu.Unlock()
+	rc, err := tc.repl.getChecksum(ctx, id)
+	if !testutils.IsError(err, "no checksum found") {
+		t.Fatal(err)
+	}
+	require.Nil(t, rc.Checksum)
+	// Next condition, the initial wait expires and checksum is not started,
+	// this will take 10ms.
+	id = uuid.FastMakeV4()
+	tc.repl.mu.Lock()
+	tc.repl.mu.checksums[id] = ReplicaChecksum{notify: make(chan struct{})}
+	tc.repl.mu.Unlock()
+	rc, err = tc.repl.getChecksum(ctx, id)
+	if !testutils.IsError(err, "checksum computation did not start") {
+		t.Fatal(err)
+	}
+	require.Nil(t, rc.Checksum)
+	// Next condition, initial wait expired and we found the started flag,
+	// so next step is for context deadline.
+	id = uuid.FastMakeV4()
+	tc.repl.mu.Lock()
+	tc.repl.mu.checksums[id] = ReplicaChecksum{notify: make(chan struct{}), started: true}
+	tc.repl.mu.Unlock()
+	rc, err = tc.repl.getChecksum(ctx, id)
+	if !testutils.IsError(err, "context deadline exceeded") {
+		t.Fatal(err)
+	}
+	require.Nil(t, rc.Checksum)
+
+	// Need to reset the context, since we deadlined it above.
+	ctx, cancel = context.WithTimeout(context.Background(), 100*time.Millisecond)
+	defer cancel()
+	// Next condition, node should quiesce.
+	tc.repl.store.Stopper().Quiesce(ctx)
+	rc, err = tc.repl.getChecksum(ctx, uuid.FastMakeV4())
+	if !testutils.IsError(err, "store quiescing") {
+		t.Fatal(err)
+	}
+	require.Nil(t, rc.Checksum)
 }

